docs/image-map.md

```diff
@@ -0,0 +1,136 @@
+# Image Map
+
+The Image Map is a cache used by CLUE to handle image loading. To understand this document it is best to review the document about [images](images.md)
+
+The key of an entry in the cache is its URL. As described in the images doc these URL can be:
+- regular http/https URLs
+- "local" assets stored in this repository
+- special URLs pointing at Firebase Storage or an object in the Firebase Realtime DB.
+
+Entries in the cache have the following properties:
+- **contentUrl** the location to download the image from. This URL might not be something the browser can handle directly. If the key of the entry is a URL that was modified before downloading, this contentUrl contains the modified version.
+- **displayUrl** a url to render the actual image, it should be a URL that can be used by the browser. For example the `src` of an `img` or the background of a div in css.
+- **width, height** the dimension of the image
+- **filename** if the image was loaded from a file, this is the name of the file
+- **status** what state the entry is in, see below
+
+Entries in the cache can have 4 status values:
+- **Storing**: `getImage` has been called with a URL and this URL is being processed to download and store the image data.
+- **ComputingDimensions**: after the image data has been stored, the image is rendered offscreen to find out its dimensions. The resulting dimensions are added as the width and height on the entry
+- **Ready**: the image has been stored successfully and the dimensions computed successfully
+- **Error**: something went wrong while storing or computing the dimensions
```

Comment:
There is some value, because we could then provide better error messages. We could also potentially render the image if the error was only on the dimensions. 

But I don't think it outweighs the complexity that it would introduce. Consumers would need to check for these two cases instead of one. And internally it would probably be confusing if we didn't treat them differently since the entry would actually be stored if the status was ErrorDimensions.